src/pipelines/aws/aws-pipeline.ts

```diff
@@ -84,10 +88,47 @@ export default class AWSPipeline implements Pipeline {
     }
   }
 
+  async getMediaConvertJobStatus(jobId: string): Promise<string | undefined> {
+    try {
+      const command = new GetJobCommand({ Id: jobId });
+      const response = await this.mediaConvert.send(command);
+      if (response.Job?.Status === JobStatus.ERROR) {
+        logger.error(`Job ${jobId} error ${response.Job?.ErrorMessage}`);
+      } else if (response.Job?.Status === JobStatus.CANCELED) {
+        logger.error(`Job ${jobId} ${response.Job?.Status}`);
+      }
+      return response.Job?.Status;
+    } catch (error) {
+      logger.error(`Error getting job ${jobId}: ${error}`);
+      return undefined;
+    }
+  }
+
+  async delay_ms(ms: number): Promise<void> {
```

Comment:
Since this function is pretty generic and not really tied to the aws-pipeline, it could make sense to move it somewhere else, `utils.ts` or something. Also for consistency the name should be `delayMs` or just `delay`.